rigging/rigs/network.py

```diff
@@ -0,0 +1,357 @@
+
+            # L2
+
+            eth_src = self._strmac(eth[:6])
+            eth_dst = self._strmac(eth[6:12])
+            #eth_type = struct.unpack("!H", eth[12:14])[0]
+
+            pkt_attrs["srcmac"] = eth_src
+            pkt_attrs["dstmac"] = eth_dst
+
+            # L3
+
+            ip = eth[14:]
+
+            ip_ver, ip_hdrlen = divmod(ip[0], 16)
+            ip_hdrlen *= 4 # 32bit incrs
+
+            if ip_ver != 4:
+                continue
+
+            ip_diffserv = ip[1]
+
+            ip_pktlen, ip_id, ip_flags, ip_ttl, ip_proto, ip_cksum = \
+                    unpack("!HHHBBH", ip[2:12])
+
+            ip_src = inet_ntoa(ip[12:16])
+            ip_dst = inet_ntoa(ip[16:20])
+
+            pkt_attrs['srcip'] = ip_src
+            pkt_attrs['dstip'] = ip_dst
+
+            # L4
+
+            if ip_proto == IPPROTO_TCP:
+                tcp = ip[ip_hdrlen:]
+                tcp_src, tcp_dst, tcp_seq, tcp_ack, tcp_hdrlen, tcp_flags, \
+                        tcp_win, tcp_chksum = unpack("!HHLLBBHH", tcp[0:18])
+
+                tcp_hdrlen = (tcp_hdrlen >> 4) * 4 # 32bit incrs
+                tcp_flags = TCP_FLAGS(tcp_flags)
+
+                pkt_attrs['srcport'] = tcp_src
+                pkt_attrs['dstport'] = tcp_dst
+
+                pkt_attrs['tcpflags'] = tcp_flags
+
+                payload = tcp[tcp_hdrlen:]
+
+                pkt_str = (f"{ip_src:>15s}:{tcp_src:<5d} ({eth_src}) -> "
+                             f"{ip_dst:>15s}:{tcp_dst:<5d} ({eth_dst}) "
+                             f"{str(tcp_flags).replace('TCP_FLAGS.', '')} {payload}")
+
+            elif ip_proto == IPPROTO_UDP:
+                udp = ip[ip_hdrlen:]
+                udp_src, udp_dst, udp_len, udp_cksum = unpack("!HHHH", udp[:8])
+
+                pkt_attrs['srcport'] = udp_src
+                pkt_attrs['dstport'] = udp_dst
+
+                payload = udp[8:] # UDP header is fixed 8 bytes
+
+                pkt_str = (f"{ip_src:>15s}:{udp_src:<5d} ({eth_src}) -> "
+                             f"{ip_dst:>15s}:{udp_dst:<5d} ({eth_dst}) {payload}")
+
+            elif ip_proto == IPPROTO_ICMP:
+                icmp = ip[ip_hdrlen:]
+                icmp_type, icmp_code, icmp_cksum, icmp_id, icmp_seq = \
+                        unpack("!BBHHH", icmp[:8])
+
+                try:
+                    icmp_code = ICMP_TYPE_CODE[icmp_type](icmp_code)
+                except:
+                    pass
+
+                try:
+                    icmp_type = ICMP_TYPES(icmp_type)
+                    # If parsing the type raises an excp, then just ignore it.
+                    pkt_attrs['icmptype'] = icmp_type
+                except:
+                    pass
+
+                pkt_str = (f"{ip_src:>15s} ({eth_src}) -> "
+                             f"{ip_dst:>15s} ({eth_dst}) "
+                             f"ICMP {icmp_type.name}")
+
+
+            pkt_attrs['payload'] = payload
+
+            self.log_debug(pkt_str)
```

Comment:
I'm hesitant on this one due to how this might blow up the logs, even if debug is not a default.

Reply:
Removed debug logs on the new PR https://github.com/TurboTurtle/rig/pull/46